api/src/main/java/org/apache/iceberg/actions/RewriteTableLocation.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.Table;
+
+public interface RewriteTableLocation extends Action<RewriteTableLocation, RewriteTableLocation.Result> {
```

Comment:
I'd try to avoid using the name `TableLocation`, which is a dedicated name for the location of the table as a table property. Given the action is to rewrite the file paths within all metadata files, is `rewriteFilePath` more suitable?

Reply:
To be more specific, it's `rewriteMetadataFilePath`?

Reply:
Mostly in metadata files. I think it has to touch position/eq delete files as well as they have the absolute file paths as well.

Reply:
@flyrain Yes it does modify position files as well. `RewriteFilePath` implies that the method rewrites a single file. Maybe `RewriteTablePath`?

Reply:
Yeah, the path rewritten could be a directory as well. I think `RewriteTablePath` is fine,  other alternative cuold be `RewriteAbsolutePath` or Just `RewritePath`.